versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5775 @@
+
+For example, file paths are always represented as strings, making the conversion from `String` to `File` obvious and unambiguous.
+
+```wdl
+String path = "/path/to/file"
+File f = path
+```
+
+The table below lists all globally valid coercions. The "target" type is the type being coerced to (this is often called the "left-hand side" or "LHS" of the coercion) and the "source" type is the type being coerced from (the "right-hand side" or "RHS").
+
+|Target Type |Source Type     |Notes/Constraints |
+|------------|----------------|------------------|
+|`File`|`String`||
+|`Float`|`Int`|May cause overflow error|
+|`Y?`|`X`|`X` must be coercible to `Y`|
+|`Array[Y]`|`Array[X]`|`X` must be coercible to `Y`|
+|`Map[X,Z]`|`Map[W,Y]`|`W` must be coercible to `X` and `Y` must be coercible to `Z`|
+|`Pair[X,Z]`|`Pair[W,Y]`|`W` must be coercible to `X` and `Y` must be coercible to `Z`|
+|`Object`|`Map[String,Y]`||
+|`Map[String,Y]`|`Object`|All object values must be coercible to `Y`|
+|`Struct`    |`Map[String,Y]`|`Map` keys must match `Struct` member names, and all `Struct` members types must be coercible from `Y`|
+|`Map[String,Y]`|`Struct`|All `Struct` members must be coercible to `Y`|
+|`Object`|`Struct`||
+|`Struct`|`Object`|`Object` keys must match `Struct` member names, and `Object` values must be coercible to `Struct` member types|
+
+Note that primitive types can always be converted to `String` using [string interpolation](#expression-placeholders-and-string-interpolation).
+
+```wdl
+Int i = 5
+String istring = "~{i}"
+```
+
+See [Expression Placeholder Coercion](#expression-placeholder-coercion) for details.
+
+###### Coercion of Optional Types
+
+A non-optional type `T` can always be coerced to an optional type `T?`, but the reverse is not true - coercion from `T?` to `T` is not allowed because the latter cannot accept `None`.
+
+This constraint propagates into compound types. For example, an `Array[T?]` can contain both optional and non-optional elements. This facilitates the common idiom `select_first([expr, default])`, where `expr` is of type `T?` and `default` is of type `T`, for converting an optional type to a non-optional type. However, an `Array[T?]` could not be passed to the [`sep`](#-string-sepstring-arraystring) function, which requires an `Array[T]`.
+
+The two exceptions where coercion from `T?` to `T` are allowed are:
+* [String concatenation in expression placeholders](#concatenation-of-optional-values)
+* [Equality and inequality comparisons](#equality-and-inequality-comparison-of-optional-types)
+
+###### Struct/Object coercion from Map
+
+`Struct`s and `Object`s can be coerced from map literals, but beware the following behavioral difference:
```

Comment:
There's no example here using `Struct` and given that `Object` will be deprecated, seems odd to do this via Object examples.

Reply:
Good point